api/v1/autostoragescale.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AutoStorageScalingSpec defines the desired state of AutoStorageScaling
+type AutoStorageScalingSpec struct {
+	// StorageClusterName is the name of the storage cluster for which the storage scaling is to be done.
+	// +kubebuilder:validation:Required
+	StorageClusterName string `json:"storageClusterName,omitempty"`
+	// DeviceClassName is the name of the device class for which the storage scaling is to be done.
+	// +kubebuilder:validation:Required
+	DeviceClassName string `json:"deviceClassName,omitempty"`
```

Comment:
why deviceClass is required? AFAIK its optional in the storagecluster creation

Reply:
its optionally but ocs operator alwyas add `ssd` if not provided by storagecluster

Reply:
I'm still confused though. The autoscaler should be autoscaling existing storageclassdevicesets, and the device class should be able to be read from the scDeviceSet being scaled, right?

Reply:
> I'm still confused though. The autoscaler should be autoscaling existing storageclassdevicesets, and the device class should be able to be read from the scDeviceSet being scaled, right?

AFAIK autoscaling will apply to the devicesets of a particular deviceclass, so we need to know the deviceclass in order to filter devicesets by deviceclass.

Reply:
During creation its an optional it should be kept optional as well, if its not specified we should treat it the same way we do during creation.

Reply:
> During creation its an optional it should be kept optional as well, if its not specified we should treat it the same way we do during creation.

You mean that if not specified it should default to `ssd` like `StorageCluster` CR, right?

Reply:
yes correct whatever ocs-operator chooses during storagecluster creation.

Reply:
Will set a default value to ssd